jfoenix/src/main/java/com/jfoenix/controls/JFXDrawer.java

```diff
@@ -1,923 +1,929 @@
+        DrawerDirection.LEFT);
+
+    /**
+     * creates empy drawer node
+     */
+    public JFXDrawer() {
+        initialize();
+
+        overlayPane.setBackground(new Background(new BackgroundFill(Color.rgb(0, 0, 0, 0.1),
+            CornerRadii.EMPTY,
+            Insets.EMPTY)));
+        overlayPane.getStyleClass().add("jfx-drawer-overlay-pane");
+        overlayPane.setOpacity(0);
+
+        sidePane.getStyleClass().add("jfx-drawer-side-pane");
+        sidePane.setBackground(new Background(new BackgroundFill(Color.rgb(255, 255, 255, 1),
+            CornerRadii.EMPTY,
+            Insets.EMPTY)));
+        sidePane.setPickOnBounds(false);
+        // causes performance issue when animating the drawer
+
+        this.getChildren().add(content);
+
+        // add listeners
+        overlayPane.setOnMouseClicked((e) -> close());
+        initListeners();
+
+        //  init size value
+        setDefaultDrawerSize(100);
+    }
+
+    private void initialize() {
+        this.getStyleClass().add(DEFAULT_STYLE_CLASS);
+    }
+
+    private void initListeners() {
+        updateDirection(directionProperty.get());
+        initTranslate.bind(Bindings.createDoubleBinding(() -> -1 * directionProperty.get()
+            .doubleValue() * defaultSizeProperty
+            .getValue() - initOffset * directionProperty.get().doubleValue(), defaultSizeProperty, directionProperty));
+
+        // add listeners to update drawer properties
+        overLayVisibleProperty().addListener((o, oldVal, newVal) -> {
+            overlayPane.setStyle(!newVal ? "-fx-background-color : transparent;" : "");
+            overlayPane.setMouseTransparent(!newVal);
+            overlayPane.setPickOnBounds(newVal);
+        });
+
+        directionProperty.addListener((o, oldVal, newVal) -> updateDirection(newVal));
+        initTranslate.addListener((o, oldVal, newVal) -> updateDrawerAnimation(newVal.doubleValue()));
+
+        // content listener for mouse hold on a side
+        this.content.addEventHandler(MouseEvent.MOUSE_PRESSED, (e) -> {
+            if (!e.isConsumed()) {
+                double size = 0;
+                long valid = 0;
+                for (int i = 0; i < callBacks.size(); i++) {
+                    if (!callBacks.get(i).call(null)) {
+                        valid++;
+                    }
+                }
+                if (directionProperty.get() == DrawerDirection.RIGHT) {
+                    size = content.getWidth();
+                } else if (directionProperty.get() == DrawerDirection.BOTTOM) {
+                    size = content.getHeight();
+                }
+
+                double eventPoint = 0;
+                if (directionProperty.get() == DrawerDirection.RIGHT || directionProperty.get() == DrawerDirection.LEFT) {
+                    eventPoint = e.getX();
+                } else {
+                    eventPoint = e.getY();
+                }
+
+                if (size + directionProperty.get()

```

Comment:
[This change was made here.](https://github.com/jfoenixadmin/JFoenix/commit/9f53564b49e1db075593ab6029ee058a28bd9203#diff-ed09c98ef323d180e40af3c82a5526c9R175) But is doesn't matter if it's not breaking functionality.